Add release notes for APIGateway 2.7 (#1340)
kyma-project/api-gateway: docs/release-notes/2.7.0.md

> The release 2.7 of the API Gateway module includes the fully functional version of APIRule `v2alpha1` with [**noAuth**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-noauth-access-strategy), [**jwt**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-jwt-access-strategy), and [**extAuth**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-extauth-access-strategy) access strategies. Version `v2alpha1` is not meant for use in a production environment but for testing purposes only. After the testing phase, this version of APIRule is expected to be promoted to stable version `v2` without significant changes.

Reviewer: ```suggestion
Release 2.7 of the API Gateway module includes the fully functional version of APIRule `v2alpha1` with [**noAuth**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-noauth-access-strategy), [**jwt**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-jwt-access-strategy), and [**extAuth**](https://kyma-project.io/#/api-gateway/user/custom-resources/apirule/v2alpha1/04-15-api-rule-access-strategies?id=configuration-of-the-extauth-access-strategy) access strategies. Version `v2alpha1` is not meant for use in a production environment but for testing purposes only. After the testing phase, this version of APIRule is expected to be promoted to stable version `v2` without significant changes.
```

> For SAP BTP, Kyma runtime, API Gateway 2.7 is scheduled to be rolled out to the regular channel on October 28, 2024. After this date, users will be required to start testing the migration procedure. To learn more, see the [APIRule Migration blog post](https://community.sap.com/t5/technology-blogs-by-sap/sap-btp-kyma-runtime-apirule-migration-noauth-and-jwt-handlers/ba-p/13882833). It provides the timeline for deletion of APIRule `v1beta1` and covers the following migration scenarios:

Reviewer: ```suggestion
For SAP BTP, Kyma runtime, API Gateway 2.7 is scheduled to be rolled out to the regular channel on October 28, 2024. After this date, users will be required to start testing the migration procedure. To learn more, see the [APIRule migration blog post](https://community.sap.com/t5/technology-blogs-by-sap/sap-btp-kyma-runtime-apirule-migration-noauth-and-jwt-handlers/ba-p/13882833). It provides the timeline for deletion of APIRule `v1beta1` and covers the following migration scenarios:
```
APIRule migration is not the exact title of the blog post so I would not capitalize it.

>     - From **noop**, **allow**, or **no_auth** handler in version `v1beta1` to the **noAuth** handler in version `v2alpha1`

Reviewer: ```suggestion
    - From the **noop**, **allow**, or **no_auth** handler in version `v1beta1` to the **noAuth** handler in version `v2alpha1`
```

    - From the **jwt** handler in version `v1beta1` to the new configuration of the **jwt** handler in version `v2alpha1`
The migration procedure from OAuth2 handlers based on Ory Oathkeeper to the new **extAuth** handler will be published in the next blog post. For now, see the tutorial [Exposing a Workload with extAuth](https://kyma-project.io/#/api-gateway/user/tutorials/01-50-expose-and-secure-a-workload/v2alpha1/01-53-expose-and-secure-workload-extauth.md).

## Deprecation

> - This release instroduces deprecatin of the APIRule custom resource in version `v1beta1`.

Reviewer: ```suggestion
- This release introduces the deprecation of the APIRule custom resource in version `v1beta1`.
```
